### ISSUE TITLE
Train Slow

### ISSUE THREAD

**Issue by FeiZhuNiU** [NVIDIA/retinanet-examples#151]
focal loss: 1.107, box loss: 0.446, 8.943s/1-batch (fw: 3.032s, bw: 5.043s), 0.1 im/s, lr: 0.00015

fw:  most consumed by   model._compute_loss()

how to accelerate?

**Comment by james-nvidia** [comment]
Please provide details of your hardware and environment setup. Which GPU are you using? Which dataset? What command did you use to start training?

**Comment by yashnv** [comment]
@FeiZhuNiU did you solve this issue?

**Comment by FeiZhuNiU** [comment]
I was not following the steps that training in docker container, I just use the model and training in normal ways.  
The phenomenon is:
the first target in the loop `for target in targets` consumed 3 seconds (strange..)
The I tried `print(targets)` outside the `_extract_targets` function, then the whole loop ran fast. However the `print(targets)` consumed 3 seconds..
```
    def _extract_targets(self, targets, stride, size):
        cls_target, box_target, depth = [], [], []
        for target in targets:
            target = target[target[:, -1] > -1]
            if stride not in self.anchors:
                self.anchors[stride] = generate_anchors(stride, self.ratios, self.scales)
            snapped = snap_to_anchors(
                target, [s * stride for s in size[::-1]], stride,
                self.anchors[stride].to(targets.device), self.classes, targets.device)
            for l, s in zip((cls_target, box_target, depth), snapped): l.append(s)
        return torch.stack(cls_target), torch.stack(box_target), torch.stack(depth)
```

I would close the issue and will follow run in container instead.

@yashnv  It would be thankful If you have any idea on this issue :)
